geopandas/geodataframe.py

```diff
@@ -1333,6 +1333,11 @@ def explode(self):
         1 0  name2  POINT (2.00000 1.00000)
           1  name2  POINT (0.00000 0.00000)
         """
+
+        if column != self.geometry.name:
+            exploded_df = pd.DataFrame(self).explode(column, **kwargs)
```

Comment:
```suggestion
            exploded_df = super().explode(column, **kwargs)
```

should normally do it as well. But looking at what you said on the issue, you ran into troubles with this?

Reply:
I think the issue is that this picks `explode` from `GeoPandasBase` not `DataFrame`. https://github.com/geopandas/geopandas/blob/c325570e86ec534857ed03fd256f2426e3b2aad0/geopandas/base.py#L1203

Reply:
We can move the `explode` method from `base` to `GeoSeries` (there's no reason to have it in base anyway since we override it in GeoDataFrame). Then `return super().explode(column, **kwargs)` works as expected.

@iboates can you do that? Take the whole `GeoPandasBase.explode` method from `base.py` and move is to `GeoSeries` class?

Reply:
Yup, the first one doesn't work. but `super(GeoPandasBase, self).explode(column, **kwargs)` apparently does work (the two new tests pass). How is this possible? `GeoPandasBase` inherits directly from `object`

Reply:
I _think_ that it does that due to MRO (Method Resolution Order). 

See `gpd.GeoDataFrame.__mro__`:

```
(geopandas.geodataframe.GeoDataFrame,
 geopandas.base.GeoPandasBase,
 pandas.core.frame.DataFrame,
 pandas.core.generic.NDFrame,
 pandas.core.base.PandasObject,
 pandas.core.accessor.DirNamesMixin,
 pandas.core.base.SelectionMixin,
 pandas.core.indexing.IndexingMixin,
 object)
```

Normal `super` uses `geopandas.base.GeoPandasBase` because it is first in the line. `super(GeoPandasBase, self)` then skips one step and uses `pandas.core.frame.DataFrame`. That is my assumption.

Reply:
oh wow this is some deep python

Reply:
I missed your comment from about an hour ago, I'll look at it over the weekend.